src/cats/match.cljc

```diff
@@ -0,0 +1,117 @@
+(ns cats.match
+  (:require
+    [cats.core :as c]
+    #?@(:clj  [[cats.monad.exception]
+               [cats.monad.maybe]
+               [cats.monad.either]
+               [clojure.core.match.protocols :refer [IMatchLookup]]]
+        :cljs [[cats.monad.exception :refer [Success Failure]]
+               [cats.monad.maybe :refer [Just Nothing]]
+               [cats.monad.either :refer [Left Right]]]))
+  #?(:clj
+     (:import (cats.monad.exception Success)
+              (cats.monad.exception Failure)
+              (cats.monad.maybe Just)
+              (cats.monad.maybe Nothing)
+              (cats.monad.either Left)
+              (cats.monad.either Right))))
+
+(defn- handle-success [this k not-found]
+  (case k
```

Comment:
Feels like there somehow could be a single function for `handle-success/failure/just`.
Anyway, I'd go with `if` for a single case. If there are multiple success cases - can be `(if (#{:success :found} ...))`.